packages/mui-material/src/Accordion/Accordion.spec.tsx

```diff
@@ -56,3 +56,29 @@ const AccordionComponentTest = () => {
 <Accordion slotProps={{ heading: { component: 'h4' } }}>
   <div />
 </Accordion>;
+
+function Custom(props: AccordionProps) {
+  const { slotProps, ...dialogProps } = props;
+  return (
+    <Accordion
+      slotProps={{
+        ...slotProps,
+        transition: (ownerState) => {
+          const transitionProps =
+            typeof slotProps?.transition === 'function'
+              ? slotProps.transition(ownerState)
+              : slotProps?.transition;
+          return {
+            ...transitionProps,
+            onExited: (node) => {
+              transitionProps?.onExited?.(node);
+            },
+          };
+        },
```

Comment:
Did users need to do the `typeof slotProps?.transition === 'function'` check before this PR?

Reply:
> Did users need to do the typeof slotProps?.transition === 'function' check before this PR?

Yes, they need to regardless of using `mergeSlotProps` util.

Reply:
But ideally, they would be using the `mergeSlotProps` util, right?